test/deal-investor/get-deal-investors-by-deal-identifier.api-test.ts

```diff
@@ -0,0 +1,112 @@
+import { UkefId } from '@ukef/helpers';
+import { withAcbsAuthenticationApiTests } from '@ukef-test/common-tests/acbs-authentication-api-tests';
+import { Api } from '@ukef-test/support/api';
+import { ENVIRONMENT_VARIABLES } from '@ukef-test/support/environment-variables';
+import { RandomValueGenerator } from '@ukef-test/support/generator/random-value-generator';
+import nock from 'nock';
+
+import { DealInvestorGenerator } from '../support/generator/deal-investor-generator';
+
+describe('GET /deals/{dealIdentifier}/deal-investors', () => {
+  const valueGenerator = new RandomValueGenerator();
+  // TODO: Get portfolio default value from default constants;
+  const dealIdentifier: UkefId = valueGenerator.ukefId();
+  const portfolioIdentifier = 'E1';
+  const getDealInvestorsUrl = `/api/v1/deals/${dealIdentifier}/investors`;
+
+  let api: Api;
+
+  const { dealInvestorsInAcbs, dealInvestorsFromApi } = new DealInvestorGenerator(valueGenerator).generate({
+    numberToGenerate: 2,
+    dealIdentifier,
+    portfolioIdentifier,
+  });
+  const expectedDealInvestors = dealInvestorsFromApi;
+
+  beforeAll(async () => {
+    api = await Api.create();
+  });
+
+  afterAll(async () => {
+    await api.destroy();
+  });
+
+  afterEach(() => {
+    nock.abortPendingRequests();
+    nock.cleanAll();
+  });
+
+  const { idToken, givenAuthenticationWithTheIdpSucceeds } = withAcbsAuthenticationApiTests({
+    givenRequestWouldOtherwiseSucceed: () => requestToGetDealInvestors().reply(200, dealInvestorsInAcbs),
+    makeRequest: () => api.get(getDealInvestorsUrl),
+  });
+
+  it('returns a 200 response with the deal investors if it is returned by ACBS', async () => {
+    givenAuthenticationWithTheIdpSucceeds();
+    requestToGetDealInvestors().reply(200, dealInvestorsInAcbs);
+
+    const { status, body } = await api.get(getDealInvestorsUrl);
+
+    expect(status).toBe(200);
+    expect(body).toStrictEqual(JSON.parse(JSON.stringify(expectedDealInvestors)));
+  });
+
+  it('returns a 404 response if ACBS returns a 200 response with the string "null"', async () => {
+    givenAuthenticationWithTheIdpSucceeds();
+    requestToGetDealInvestors().reply(200, 'null');
+
+    const { status, body } = await api.get(getDealInvestorsUrl);
+
+    expect(status).toBe(404);
+    expect(body).toStrictEqual({
+      statusCode: 404,
+      message: 'Not found',
+    });
+  });
+
+  it('returns a 500 response if ACBS returns a 200 response without the string "null"', async () => {
```

Comment:
I'd be more specific here (to distinguish this case from the case where ACBS returns the data correctly) e.g., 'returns a 500 response if ACBS returns a 200 response with a string other than "null"'